operators/utilities.py

```diff
@@ -201,3 +205,25 @@ def select_target_ob(context, sketch):
     if target_ob.name in context.view_layer.objects:
         target_ob.select_set(True)
         context.view_layer.objects.active = target_ob
+
+
+@contextmanager
+def safe_constraints(context: Context, sketch: SlvsSketch = None, constraints: SlvsConstraints = None):
+    sketch = sketch or context.scene.sketcher.active_sketch
+    constraints = constraints or context.scene.sketcher.constraints
+
+    solve = solve_system(context, sketch)
```

Comment:
We can probably avoid the initial solve check and just get the value from sketch.solver_state

Reply:
Oh I wasn't aware of that, I'll use it instead of calling the function then yeah.